pkg/cmd/inspect/inspect.go

```diff
@@ -19,30 +19,53 @@ package inspect
 import (
 	"context"
 	"fmt"
+	"kitops/pkg/artifact"
 	"kitops/pkg/lib/constants"
 	"kitops/pkg/lib/repo"
 
+	"github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-func inspectReference(ctx context.Context, opts *inspectOptions) (*ocispec.Manifest, error) {
+// Utility struct for formatting output of inspect
+type inspectInfo struct {
+	Digest     digest.Digest     `json:"digest,omitempty" yaml:"digest,omitempty"`
+	CLIVersion string            `json:"cliVersion,omitempty" yaml:"cliVersion,omitempty"`
+	Kitfile    *artifact.KitFile `json:"kitfile,omitempty" yaml:"kitfile,omitempty"`
+	Manifest   *ocispec.Manifest `json:"manifest,omitempty" yaml:"manifest,omitempty"`
+}
+
+func inspectReference(ctx context.Context, opts *inspectOptions) (*inspectInfo, error) {
 	if opts.checkRemote {
 		return getRemoteManifest(ctx, opts)
 	} else {
 		return getLocalManifest(ctx, opts)
 	}
 }
 
-func getLocalManifest(ctx context.Context, opts *inspectOptions) (*ocispec.Manifest, error) {
+func getLocalManifest(ctx context.Context, opts *inspectOptions) (*inspectInfo, error) {
 	storageRoot := constants.StoragePath(opts.configHome)
 	store, err := repo.NewLocalStore(storageRoot, opts.modelRef)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read local storage: %w", err)
 	}
-	return repo.ResolveManifest(ctx, store, opts.modelRef.Reference)
+	desc, manifest, config, err := repo.ResolveManifestAndConfig(ctx, store, opts.modelRef.Reference)
+	if err != nil {
+		return nil, err
+	}
+	version := "unknown"
+	if manifest.Annotations != nil && manifest.Annotations[constants.CliVersionAnnotation] != "" {
+		version = manifest.Annotations[constants.CliVersionAnnotation]
+	}
+	return &inspectInfo{
+		Digest:     desc.Digest,
+		CLIVersion: version,
+		Kitfile:    config,
+		Manifest:   manifest,
+	}, nil
 }
 
-func getRemoteManifest(ctx context.Context, opts *inspectOptions) (*ocispec.Manifest, error) {
+func getRemoteManifest(ctx context.Context, opts *inspectOptions) (*inspectInfo, error) {
```

Comment:
Is this repeating too much of the `getLocalManifest`? perhps we need to refactor this logic after repositroy creation to a function.